### PR TITLE
Add Kestrel keep alive ping section

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/dotnet/AspNetCore.Docs/issues/20180
Fixes https://github.com/dotnet/aspnetcore/issues/26247

### CODE REVIEW

File: aspnetcore/fundamentals/servers/kestrel.md [hunk 1]
```diff
@@ -340,6 +340,26 @@ webBuilder.ConfigureKestrel(serverOptions =>
 
 The default value is 96 KB (98,304).
 
+### HTTP/2 keep alive ping configuration
+
+Kestrel can be configured to send HTTP/2 pings to connected clients. HTTP/2 pings serve multiple purposes:
+
+* Keep idle connections alive. Some clients and proxy servers will close connections that are idle. HTTP/2 pings are considered as activity on a connection, and will prevent the connection from being closed as idle.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/kestrel.md:347]:
```suggestion
* Keep idle connections alive. Some clients and proxy servers will close connections that are idle. HTTP/2 pings are considered as activity on a connection, and prevent the connection from being closed as idle.
```

File: aspnetcore/fundamentals/servers/kestrel.md [hunk 2]
```diff
@@ -340,6 +340,26 @@ webBuilder.ConfigureKestrel(serverOptions =>
...
+* Close unhealthy connections. Connections where the client doesn't response to the keep alive ping in the configured time will be closed by the server.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/kestrel.md:348]:
```suggestion
* Close unhealthy connections. Connections where the client doesn't respond to the keep alive ping in the configured time are closed by the server.
```

File: aspnetcore/fundamentals/servers/kestrel.md [hunk 3]
```diff
@@ -340,6 +340,26 @@ webBuilder.ConfigureKestrel(serverOptions =>
...
+
+There are two configuration options related to HTTP/2 keep alive pings:
+
+* `Http2.KeepAlivePingInterval` is a `TimeSpan` that configures the ping internal. The server will send a keep alive ping to the client if it doesn't receive any frames for this period of time. Keep alive pings are disabled when this option is set to `TimeSpan.MaxValue`. The default value is `TimeSpan.MaxValue`.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/kestrel.md:352]:
```suggestion
* `Http2.KeepAlivePingInterval` is a `TimeSpan` that configures the ping internal. The server sends a keep alive ping to the client if it doesn't receive any frames for this period of time. Keep alive pings are disabled when this option is set to `TimeSpan.MaxValue`. The default value is `TimeSpan.MaxValue`.
```

File: aspnetcore/fundamentals/servers/kestrel.md [hunk 4]
```diff
@@ -340,6 +340,26 @@ webBuilder.ConfigureKestrel(serverOptions =>
...
+* `Http2.KeepAlivePingTimeout` is a `TimeSpan` that configures the ping timeout. If the server doesn't receive any frames, such as a response ping, during this timeout then the connection will be closed. Keep alive timeout is disabled when this option is set to `TimeSpan.MaxValue`. The default value is 20 seconds.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/kestrel.md:353]:
```suggestion
* `Http2.KeepAlivePingTimeout` is a `TimeSpan` that configures the ping timeout. If the server doesn't receive any frames, such as a response ping, during this timeout then the connection is closed. Keep alive timeout is disabled when this option is set to `TimeSpan.MaxValue`. The default value is 20 seconds.
```

**Review by Rick-Anderson — APPROVED** [review]
Looks good. Per style guide, avoid future tense.

**Review by halter73 — APPROVED** [review]

**Comment by guardrex** [comment]
This PR broke the topic's versioning. Moniker ranges can't be nested, so what one has to do is break in and out of whatever range is there for the new content. In this case, the addition will be .....

```
... EXISTING CONTENT ...

::: moniker-end

::: moniker range=">= aspnetcore-5.0"

... NEW CONTENT

::: moniker-end

::: moniker range=">= aspnetcore-3.0"

... EXISTING CONTENT ...
```

I'll patch this up now and take it live immediately.

**Pull request merged**